asm/f32/benchDot_test.go

```diff
@@ -0,0 +1,89 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build go1.7
+
+package f32
+
+import (
+	"fmt"
+	"testing"
+)
+
+func BenchmarkDotUnitary(t *testing.B) {
+	const name = "DotUnitary"
+	for _, v := range []int64{1, 2, 3, 4, 5, 10, 100, 1e3, 5e3, 1e4, 5e4} {
+		t.Run(fmt.Sprintf("%s-%d", name, v), func(b *testing.B) {
+			x, y := x[:v], y[:v]
+			b.SetBytes(128 * v)
+			for i := 0; i < b.N; i++ {
+				_ = DotUnitary(x, y)
```

Comment:
`_ = func()` is the same as `func()`, so if the compiler is going to optimise this away, the `_` makes no difference. You need to assign to a package global.